src/core/server/saved_objects/saved_objects_type_registry.ts

```diff
@@ -72,18 +72,15 @@ export class SavedObjectTypeRegistry {
    * resolves to `false` if the type is not registered
    */
   public isNamespaceAgnostic(type: string) {
-    return (
-      this.types.get(type)?.namespaceType === 'agnostic' ||
-      this.types.get(type)?.namespaceAgnostic ||
-      false
-    );
+    return this.types.get(type)?.namespaceType === 'agnostic';
   }
 
   /**
    * Returns whether the type is single-namespace (isolated);
    * resolves to `true` if the type is not registered
    */
   public isSingleNamespace(type: string) {
+    // in the case we somehow registered a type with an invalid `namespaceType`, treat it as single-namespace
     return !this.isNamespaceAgnostic(type) && !this.isMultiNamespace(type);
```

Comment:
Not sure this is really necessary but I'm fine with it.

Reply:
I actually ran into this with some of our unit tests that are still plain JavaScript. Figured it was better to be safe than sorry!